src/nrniv/singlech.cpp

```diff
@@ -163,7 +163,8 @@ void SingleChan::set_rates(double v) {
             state_[i].n_ = 0;
         }
         current_chan = (SingleChan*) this;
-        (*info_->f_)(v, nprop_->prop()->param, nprop_->prop()->dparam);
+        assert(false);
```

Comment:
This line also does not seem to be reached.

Reply:
cc: @nrnhines

Reply:
I will work on tests for the commented out fragments for kschan.cpp, shape.cpp, and singlech.cpp.
But as an initial observation, note that neurondemo crashes if one chooses "Linear Circuit Dynamic Clamp" and then
chooses "Stochastic Single Channels: HH". Also, for either of those models. "Init&Run" does not produce valid results.
By the way, both those models use kschan.cpp and Stochastic also uses singlech.cpp

Reply:
I was mistaken about singlech.cpp. It is kssingle.cpp that is used for kschan.cpp stochastic channels. It is likely that
singlech.cpp (the HOC SingleChan class) will be removed from NEURON as it is a special case of the KSChan stochastic channel.

After executing the tests in nrn/external/nrntest/nrniv/KSChan with
```
for i in *.hoc *.ses ; do nrngui $i -c 'run()' -c 'quit()' ; done
```
The coverage of kschan.cpp reaches 1586 of 2397 lines but, unfortunately, that coverage misses the commented out fragment.  So I need to write a special test to cover it.

Reply:
#2082 removes SingleChan from NEURON. All of its functionality is subsumed under KSChan.